src/tools.ts

```diff
@@ -41,8 +42,26 @@ function detectInfectionCommand(): string {
     return 'phpdbg -qrr ./vendor/bin/infection';
 }
 
-export default function createTools(config: Config): Array<Tool> {
-    return [
+function backwardCompatibilityCheckTool(action: Action): Tool | null {
+    const baseSha1 = action.getBaseBranchSha1();
+
+    if (baseSha1 === null) {
+        return null;
+    }
+
+    return {
+        // @TODO need to `git fetch baseSha1` from source repo!
+        executionType : ToolExecutionType.STATIC,
+        name          : 'Backward Compatibility Check',
+        command       : `roave-backward-compatibility-check check --from="${ baseSha1 }" --install-development-dependencies`,
+        filesToCheck  : [ 'composer.json' ],
+        toolType      : ToolType.CODE_CHECK,
```

Comment:
Defining `before_script` here would probably be needed.

Reply:
Have you tried adding `git fetch ${baseSha1}` as a `before_script`?

Reply:
How do we add a `before_script` though?

Reply:
ah yes, at this point I'd added `lintConfigCommand` because the file that exists in `filesToCheck` is appended to it before it gets added to `before_script`.

You could add another option key `beforeScript`, and then modify https://github.com/laminas/laminas-ci-matrix-action/blob/3e17f6852334fef66e6b973a0b263cbacaa94458/src/config/app.ts#L290 to use it?

Reply:
Be careful as I did that originally but needed a way to append the filename to the lint command.

Reply:
https://github.com/laminas/laminas-ci-matrix-action/pull/126#discussion_r1256601809